frostsnap_coordinator/src/bitcoin/wallet.rs

```diff
@@ -330,37 +210,19 @@ impl FrostsnapWallet {
         &mut self,
         update: spk_client::SyncResult<ConfirmationTimeHeightAnchor>,
     ) -> Result<bool> {
-        let indexed_tx_graph_changeset = self.graph.apply_update(update.graph_update);
-        let chain_changeset = self.chain.apply_update(update.chain_update)?;
-        self.db.lock().unwrap().execute(|tx| {
-            let chain_list = self.chain_list_handle.api(&tx);
-            let changed = !chain_changeset.is_empty() || !indexed_tx_graph_changeset.is_empty();
-
-            if !chain_changeset.is_empty() {
-                chain_list.push(&bincode::serde::Compat(chain_changeset))?;
-            }
-
-            self.consume_indexed_tx_graph_changeset(tx, indexed_tx_graph_changeset)?;
-
-            Ok(changed)
-        })
-    }
-
-    fn consume_indexed_tx_graph_changeset(
-        &self,
-        tx: &mut llsdb::Transaction<impl llsdb::Backend>,
-        mut changeset: WalletIndexedTxGraphChangeSet,
-    ) -> Result<()> {
-        // We never want to add keychain descriptors to the database. We derive them from the keys.
-        changeset.indexer.keychains_added.clear();
-        if !changeset.graph.is_empty() {
-            let tx_graph_list = self.tx_graph_list_handle.api(&tx);
-            tx_graph_list.push(&bincode::serde::Compat(changeset.graph))?;
-        }
-
-        self.consume_spk_revelation_change(tx, changeset.indexer)?;
-
-        Ok(())
+        let mut db = self.db.lock().unwrap();
+
+        let changed =
+            self.tx_graph
+                .multi(&mut self.chain)
```

Comment:
These `Multi` types are so we can mutate tuple tx_graph and chain within the same db call?

Reply:
Yes. This will be gone in the future hopefully.